### PR TITLE
update feeding_data.rst doc

### PR DESCRIPTION
**Pull request by JepsonWong**
update feeding_data.rst doc.

image
image

### CODE REVIEW

File: doc/fluid/advanced_guide/data_preparing/feeding_data_en.rst [hunk 1]
```diff
@@ -4,31 +4,31 @@
...
+In the code above, :code:`image` and :code:`label` are two input data layers created by :code:`fluid.data` . :code:`image` is float data of shape :code:`[3, 224, 224]` ; :code:`label` is the int data of shape :code:`[1]` . Note that:
 
-1. :code:`-1` is represented for the dimension of batch size by default in Fluid. And :code:`-1` is added to the first dimension of :code:`shape` by default. Therefore in the code above, it would be alright to transfer numpy array of :code:`[32, 3, 224, 224]` to :code:`image` . If you want to customize the position of the batch size dimension, please set :code:`fluid.layers.data(append_batch_size=False)` .Please refer to the tutorial in the advanced user guide: :ref:`user_guide_customize_batch_size_rank_en` .
+1. When the Executor executes, it checks whether the :code:`shape` and :code:`dtype` defined in the data layer's data and feed data are consistent. If they are not consistent, the program will exit with an error. For some tasks, in different rounds, some dimensions of the data will change, You can set the dimension value to None. For example, if the 0th dimension will change, you can set :code:`shape` to :code:`[None , 100]`.
```
Comment (phlrain) [inline doc/fluid/advanced_guide/data_preparing/feeding_data_en.rst]: 1. When the program is executing, executor will check whether the code: `shape` and code:` dtype` defined and feeded are consistent. If they are not consistent, the program will exit with an error. In some tasks, the dimension will change in different training steps. For this case, the value of the dimension can be set to None. For example, the code: `shape` can be set to code:` [None, 100] ` when the 0th dimension will change.

Comment (JepsonWong) [reply doc/fluid/advanced_guide/data_preparing/feeding_data_en.rst]: done

File: doc/fluid/advanced_guide/data_preparing/feeding_data.rst [hunk 2]
```diff
@@ -4,37 +4,33 @@
...
-通过 :code:`fluid.layers.data()` 可以配置神经网络中需要的数据层。具体方法为:
+通过 :code:`fluid.data()` 可以配置神经网络中需要的数据层。具体方法为:
 
 .. code-block:: python
 
    import paddle.fluid as fluid
 
-   image = fluid.layers.data(name="image", shape=[3, 224, 224])
-   label = fluid.layers.data(name="label", shape=[1], dtype="int64")
+   image = fluid.data(name="image", shape=[3, 224, 224])
```
Comment (phlrain) [inline doc/fluid/advanced_guide/data_preparing/feeding_data.rst]: [3, 224, 224] -> [None, 3, 224, 224]

File: doc/fluid/advanced_guide/data_preparing/feeding_data.rst [hunk 3]
```diff
@@ -4,37 +4,33 @@
...
+   label = fluid.data(name="label", shape=[1], dtype="int64")
```
Comment (phlrain) [inline doc/fluid/advanced_guide/data_preparing/feeding_data.rst]: [1] - > [None, 1]

File: doc/fluid/advanced_guide/data_preparing/feeding_data.rst [hunk 4]
```diff
@@ -4,37 +4,33 @@
...
 
    # use image/label as layer input
    prediction = fluid.layers.fc(input=image, size=1000, act="softmax")
    loss = fluid.layers.cross_entropy(input=prediction, label=label)
    ...
 
-上段代码中，:code:`image` 和 :code:`label` 是通过 :code:`fluid.layers.data`
+上段代码中，:code:`image` 和 :code:`label` 是通过 :code:`fluid.data`
 创建的两个输入数据层。其中 :code:`image` 是 :code:`[3, 224, 224]` 维度的浮点数据;
```
Comment (phlrain) [inline doc/fluid/advanced_guide/data_preparing/feeding_data.rst]: [None, 3, 224, 224],  [None, 1]

File: doc/fluid/advanced_guide/data_preparing/feeding_data.rst [hunk 5]
```diff
@@ -79,7 +75,7 @@ PaddlePaddle Fluid支持使用 :code:`fluid.layers.data()` 配置数据层；
 
 .. code-block:: python
 
-   sentence = fluid.layers.data(name="sentence", dtype="int64", shape=[1], lod_level=1)
+   sentence = fluid.data(name="sentence", dtype="int64", shape=[1], lod_level=1)
```
Comment (phlrain) [inline doc/fluid/advanced_guide/data_preparing/feeding_data.rst]: [None, 1]

File: doc/fluid/advanced_guide/data_preparing/feeding_data.rst [hunk 6]
```diff
@@ -4,37 +4,33 @@
...
+上段代码中，:code:`image` 和 :code:`label` 是通过 :code:`fluid.data`
 创建的两个输入数据层。其中 :code:`image` 是 :code:`[3, 224, 224]` 维度的浮点数据;
 :code:`label` 是 :code:`[1]` 维度的整数数据。这里需要注意的是:
 
-1. Fluid中默认使用 :code:`-1` 表示 batch size 维度，默认情况下会在 :code:`shape`
-   的第一个维度添加 :code:`-1` 。 所以 上段代码中， 我们可以接受将一个
-   :code:`[32, 3, 224, 224]` 的numpy array传给 :code:`image` 。 如果想自定义batch size
-   维度的位置的话，请设置 :code:`fluid.layers.data(append_batch_size=False)` 。
-   请参考进阶使用中的 :ref:`user_guide_customize_batch_size_rank` 。
+1. Executor在执行的时候，会检查定义的数据层数据和feed的数据的 :code:`shape` 和 :code:`dtype` 是否一致，如果不一致，程序会报错退出。对于一些任务，在不同的轮数，数据的某些维度会变化，可以将维度的值设置为None，例如第0维会变化，可以将 :code:`shape` 设置为 :code:`[None, 100]` 。
```
Comment (phlrain) [inline doc/fluid/advanced_guide/data_preparing/feeding_data.rst]: [None, 100] 改成 [None, 3, 224, 224]

File: doc/fluid/advanced_guide/data_preparing/feeding_data.rst (hunk 2 above)
Comment (JepsonWong) [reply doc/fluid/advanced_guide/data_preparing/feeding_data.rst]: done

File: doc/fluid/advanced_guide/data_preparing/feeding_data.rst (hunk 3 above)
Comment (JepsonWong) [reply doc/fluid/advanced_guide/data_preparing/feeding_data.rst]: done

File: doc/fluid/advanced_guide/data_preparing/feeding_data.rst (hunk 4 above)
Comment (JepsonWong) [reply doc/fluid/advanced_guide/data_preparing/feeding_data.rst]: done

File: doc/fluid/advanced_guide/data_preparing/feeding_data.rst (hunk 6 above)
Comment (JepsonWong) [reply doc/fluid/advanced_guide/data_preparing/feeding_data.rst]: done

File: doc/fluid/advanced_guide/data_preparing/feeding_data.rst (hunk 5 above)
Comment (JepsonWong) [reply doc/fluid/advanced_guide/data_preparing/feeding_data.rst]: done

**Review by phlrain — APPROVED** [review]

**Review by swtkiwi — APPROVED** [review]
LGTM

**Pull request merged**